### PR TITLE
Call cleanupQuery for queries that are canceled

### PR DESCRIPTION
**Pull request by nezihyigitbasi**
SqlQueryExecution::start() registers the metadata in MetadataManager and
adds to the catalogsByQueryId map. However, the metadata entry is not removed
for canceled queries (only called for finished/failed queries), which results
in a leak of metadata objects.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java [hunk 1]
```diff
@@ -672,6 +660,16 @@ public boolean transitionToCanceled()
         return canceled;
     }
 
+    private void cleanupQuery()
```
Comment (haozhun) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java]: maybe call this `cleanupQueryQuietly`?

File: presto-tests/src/test/java/com/facebook/presto/tests/TestMetadataManager.java [hunk 2]
```diff
@@ -71,4 +79,30 @@ public void testMetadataIsClearedAfterQueryFailed()
...
+    public void testMetadataIsClearedAfterQueryCanceled()
+            throws Exception
+    {
+        ExecutorService queryExecutor = Executors.newSingleThreadExecutor();
+        DistributedQueryRunner distributedQueryRunner = createQueryRunner();
+        MetadataManager metadata = (MetadataManager) distributedQueryRunner.getMetadata();
+        TestingPrestoServer coordinator = distributedQueryRunner.getCoordinator();
+
+        // submit a relatively long running query
+        queryExecutor.submit(() -> distributedQueryRunner.execute("SELECT COUNT(*) FROM lineitem"));
```
Comment (dain) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestMetadataManager.java]: I am concerned that this query could finish before the query can be canceled.  Instead I would do something like
```java
        DistributedQueryRunner distributedQueryRunner = createQueryRunner();
        QueryManager queryManager = distributedQueryRunner.getCoordinator().getQueryManager();
        MetadataManager metadata = (MetadataManager) distributedQueryRunner.getMetadata();

        QueryId queryId = queryManager.createQuery(new TestingSessionFactory(TEST_SESSION), "SELECT * FROM lineitem").getQueryId();

        // wait until query starts running                                        
        while (true) {
            if (queryManager.getQueryInfo(queryId).getState() != RUNNING) {
                break;
            }
            Thread.sleep(100);
        }

        // cancel query
        queryManager.cancelQuery(queryId);
        assertEquals(metadata.getCatalogsByQueryId().size(), 0);
```

Comment (nezihyigitbasi) [inline presto-tests/src/test/java/com/facebook/presto/tests/TestMetadataManager.java]: yep, looks better.

**Pull request merged**